include/plugin_sync.hpp

```diff
@@ -0,0 +1,134 @@
+// distribution.
+//
+// (3)The name of the author may not be used to
+// endorse or promote products derived from this software without
+// specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+// WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
+// INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+#include <atomic>
+#include <cstdint>
+#include <cstdlib>
+#include <fcntl.h>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <map>
+#include <mutex>
+#include <numeric>
+#include <regex>
+#include <set>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <thread>
+#include <unistd.h>
+#include <vector>
+
+#include <sys/syscall.h>
+#include <sys/types.h>
+
+#include <scorep/SCOREP_MetricTypes.h>
+#include <scorep/plugin/plugin.hpp>
+
+#include <occ.hpp>
+#include <occ_sensor_t.hpp>
+#include <occ_util.hpp>
+#include <util.hpp>
+
+using scorep::plugin::log::logging;
+
+class ibmpowernv_sync_plugin
+    : public scorep::plugin::base<ibmpowernv_sync_plugin,
+                                  scorep::plugin::policy::sync_strict,
+                                  scorep::plugin::policy::scorep_clock,
+                                  scorep::plugin::policy::per_thread,
+                                  scorep::plugin::policy::synchronize,
+                                  occ_sensor_policy> {
+public:
+    ibmpowernv_sync_plugin();
+
+    std::vector<scorep::plugin::metric_property> get_metric_properties(const std::string& pattern);
+    void add_metric(const occ_sensor_t& sensor);
+    template <typename Proxy>
+    void get_current_value(const occ_sensor_t& sensor, Proxy& p);
+    void synchronize(bool is_responsible, SCOREP_MetricSynchronizationMode sync_mode);
+
+private:
+    /// indicates that a fatal error has occured and that all operation should be seized
+    std::atomic<bool> fatal_occured = false;
+
+    /// minimal duration between measurments. Shorter request use cached data
+    std::chrono::nanoseconds min_measurment_interval;
+
+    /// file descriptor for the opened occ_inband_sensors file
+    int occ_file_fd;
+    /// all sensors
+    std::set<occ_sensor_t> requested_sensors_set;
+
+    // initial measurement data
+    std::map<occ_sensor_t, all_sample_data> init_measurement_data;
+    std::chrono::steady_clock::time_point init_measurement =
```

Comment:
when you start a measurement, you do not have a result. E.g. you can only say anything about the passed energy, which is `average power since strat * time`. For the time you need `init_measurment`.